clients/source.go

```diff
@@ -40,22 +43,61 @@ func WithSourceLogger(logger zerolog.Logger) func(*SourceClient) {
 	}
 }
 
-// NewSourceClient connect to gRPC server running source plugin and returns a new SourceClient
-func NewSourceClient(cc grpc.ClientConnInterface) *SourceClient {
-	return &SourceClient{
-		pbClient: pb.NewSourceClient(cc),
+func WithSourceDirectory(directory string) func(*SourceClient) {
+	return func(c *SourceClient) {
+		c.directory = directory
 	}
 }
 
-// NewManagedSourceClient starts a new source plugin process, connects to it via gRPC server
-// and returns a new SourceClient
-func NewManagedSourceClient(ctx context.Context, path string, opts ...SourceClientOption) (*SourceClient, error) {
+func WithSourceGrpcConn(userConn *grpc.ClientConn) func(*SourceClient) {
```

Comment:
```suggestion
func WithSourceGRPCConnection(userConn *grpc.ClientConn) func(*SourceClient) {
```

Reply:
I must be missing something: why don't destinations have this option?

Reply:
It wasn't really used, it is mainly used in tests when I wanted to create a special grpc conn which is in memory. prob don't have this test yet for destinations.

Reply:
Adding a test as well?